### ISSUE TITLE
Cache Disk Problem

### ISSUE THREAD

**Issue by Lee827** [rs/SDWebImage#2257]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

1. maxCacheAge
I want to set no expiration date.
SDImageCache.shared().config.maxCacheAge = -1;
This is code is right?

2. maxCacheSize
Default 0, It is no limit?

Thank You!!!!!!

**Comment by dreampiggy** [comment]
@Lee827 

Well. Base on current implementation, I can tell the usage about your question :)

+ You should set `maxCacheAge` to `NSIntegerMax` to mean no expiration date.
+ You should set `maxCacheSize` to `0` to means no cache size limit.

**Comment by Lee827** [comment]
Thank you!!!

- And I have a new question.
- I want to synchronous download images and cache the image to disk, because my project needs download image first, then show the home view.

SDWebImageManager.shared().diskImageExists(for: imageURL, completion: { (isExist) in
         guard isExist != true else { return; } 
         SDWebImageDownloader.shared().downloadImage(with: imageURL, options: SDWebImageDownloaderOptions(rawValue: 0), progress: nil, completed: { (image, data, error, finished) in
                guard let image = image, let data = data else { return; }
                SDImageCache.shared().store(image, imageData: data, forKey: imageURL.absoluteString, toDisk: true, completion: nil);
              })
            })

- Check disk Image Exists first, then download the image. Finally, cache the image to disk.
- Loop this code to download the images.
- How to synchronous and confirm download all image?

Thank you!!!!

**Comment by dreampiggy** [comment]
Well. It's not recommend to `synchronous download` on main queue. :)

Maybe you means to prefetch the image before it actual been used. You should better check `SDWebImagePrefetcher`. Which allow you to provide a list of URLs to fetch and cache on image cache.

In fact, `SDWebImagePrefetcher` use the `SDWebImageManager`, which already handle the management between cache and downloader. You don't need to write yourself.
